### ISSUE TITLE
gce.py dynamic inventory fails

### ISSUE THREAD

**Issue by soupdiver** [ansible/ansible#11245]
I followed the guide [here](http://docs.ansible.com/guide_gce.html#gce-dynamic-inventory)

I copied the `gce.ini` and put it my  `gce_service_account_email_address`, `gce_service_account_pem_file_path` and `gce_project_id`

Everything without `'` or `"`.

I installed libcloud via pip:

``` bash
pip show apache-libcloud

---
Name: apache-libcloud
Version: 0.17.0
Location: /Library/Python/2.7/site-packages
Requires: backports.ssl-match-hostname
```

When I run `gce.py`

``` bash
./gce.py
Traceback (most recent call last):
  File "./gce.py", line 297, in <module>
    GceInventory()
  File "./gce.py", line 111, in __init__
    self.driver = self.get_gce_driver()
  File "./gce.py", line 187, in get_gce_driver
    gce = get_driver(Provider.GCE)(*args, **kwargs)
  File "/Library/Python/2.7/site-packages/libcloud/compute/drivers/gce.py", line 887, in __init__
    super(GCENodeDriver, self).__init__(user_id, key, **kwargs)
  File "/Library/Python/2.7/site-packages/libcloud/compute/base.py", line 651, in __init__
    api_version=api_version, **kwargs)
  File "/Library/Python/2.7/site-packages/libcloud/common/base.py", line 1013, in __init__
    self.connection = self.connectionCls(*args, **conn_kwargs)
  File "/Library/Python/2.7/site-packages/libcloud/compute/drivers/gce.py", line 75, in __init__
    **kwargs)
  File "/Library/Python/2.7/site-packages/libcloud/common/google.py", line 627, in __init__
    self.token_info = self.auth_conn.get_new_token()
  File "/Library/Python/2.7/site-packages/libcloud/common/google.py", line 516, in get_new_token
    return self._token_request(request)
  File "/Library/Python/2.7/site-packages/libcloud/common/google.py", line 352, in _token_request
    raise GoogleAuthError('Invalid authorization response, please '
libcloud.common.google.GoogleAuthError: 'Invalid authorization response, please check your credentials.'
```

When I also specify `GCE_INI_PATH=.gce.ini`

``` bash
GCE_INI_PATH=.gce.ini ./gce.py
Traceback (most recent call last):
  File "./gce.py", line 297, in <module>
    GceInventory()
  File "./gce.py", line 111, in __init__
    self.driver = self.get_gce_driver()
  File "./gce.py", line 187, in get_gce_driver
    gce = get_driver(Provider.GCE)(*args, **kwargs)
  File "/Library/Python/2.7/site-packages/libcloud/compute/drivers/gce.py", line 878, in __init__
    raise ValueError('Project name must be specified using '
ValueError: Project name must be specified using "project" keyword.
```

I have no real idea what is going wrong. Any ideas?

**Comment by bcoca** [comment]
# List Information

Hi!

Thanks very much for your interest in Ansible.  It sincerely means a lot to us. 

This appears to be a user question, and we'd like to direct these kinds of things to either the mailing list or the IRC channel.
- IRC: #ansible on irc.freenode.net  
- mailing list: https://groups.google.com/forum/#!forum/ansible-project

If you can stop by there, we'd appreciate it.  This allows us to keep the issue tracker for bugs, pull requests, RFEs and the like.

Thank you once again and we look forward to seeing you on the list or IRC.  Thanks!

**Comment by soupdiver** [comment]
@bcoca what do you mean with "user question"?
The plugin is from the official Ansible repository and I followed the official docs which results in errors. Is that not something which belongs to here?

**Comment by bcoca** [comment]
i was pointing you to the list because people should be able to help you get this working much easier, to your point the guide does leave some steps out, but the error messages you get seem to point to the correct direction. Your first error message was about missing credentials, which you fixed by pointing at your ini file, the 2nd is looking for a project id (which should also be in your ini file). 

Most other issues should get an appropriate error message pointing you in the right direction, if not that is also a bug. Keeping this ticket as a note to make the guide more helpful.

cc @erjohnso

**Comment by soupdiver** [comment]
@bcoca Ok I found the solution. I had entered the wrong email address. I had to take the special one from the OAuth Login.

 :expressionless:

Thanks anyway! :)

**Comment by pavananms** [comment]
Hi @soupdiver ,
I have the same problem and I am using google compute Service account ID obtained from 
Permissions -> Service accounts.
Can you let me know what e-mail address to use and how did you obtain it from console

edit:By the way, I can execute ./gce.py  correctly , seems like a separate issue

**Comment by spaceneenja** [comment]
Noob here. Having this same issue. `./gce.py --list` works but then `GCE_INI_PATH=~/.gce.ini ansible all -i gce.py -m setup` fails. IDK where to go from here.

EDIT: lol... just needed to specify the correct path to gce.ini rather than copy pasting from the docs

**Comment by hawksight** [comment]
I got a bit stuck on this, so for anyone else who has the same issue.
(Note - this is a decent guide to getting started: https://medium.com/vimeo-engineering-blog/orchestrating-gce-instances-with-ansible-d825a33793cd)

* do not use any quotes
* JSON keyfile is absolutely fine despite what the script says in the instructions.

An example is here below:
```
gce_service_account_email_address = <SERVICE_ACCOUNT_USER>@<PROJECT_ID>.iam.gserviceaccount.com
gce_service_account_pem_file_path = <PATH>/<KEYFILE>.json
gce_project_id = <PROJECT_ID>
gce_zone = <GCE_ZONE>
```

Replace the `<>` vars with your settings:
* service account email from: `gcloud iam service-accounts list`
* Use a full path
* project id - `gcloud projects list`
* zone can be any from: `gcloud compute zones list`, eg. asia-east1-c

if you actually want to validate the output, then i suggest you run:
`./gce.py --list --pretty`

To make it easy on your eyes.

**Comment by cyrilsebastian** [comment]
Hi,

I am able to run the `./gce.py --list --pretty`, but when I run the ansible ping command or playbook it gives me an error saying host not found. Were you able to figure that out?

**Comment by hawksight** [comment]
@cyrilsebastian - I hadn't tried it but I just went back and tried it. it worked for me if you following the example in the medium blog link.

eg.. it says...
```
ansible -i ~/git/ansible/inventory tag_one -m ping
```

I tried
```
ansible -i inventory tag_<some_tag_on_an_instance> -m ping
```

Note my folder structure is my:
```
ansible
   |- inventory
       |- gce.py
   |- gce.ini
```
I run command from ansible dir, hence you may need to change where your inventory is on the `-i` flag.

**Comment by cyrilsebastian** [comment]
Thanks for reply. Even I followed the same blog so my folder structure is also the same.
```
$ tree ~/git/ansible/
/home/cyril/git/ansible/
├── gce.ini
├── inventory
│   └── gce.py
└── secrets.py
```

But when I run `ansible -i ~/git/ansible/inventory instance_name -m ping`, I am getting an unreachable error.
```
hostname1 | UNREACHABLE! => {
    "changed": false, 
    "msg": "ERROR! SSH encountered an unknown error during the connection. We recommend you re-run the command using -vvvv, which will enable SSH debugging output to help diagnose the issue", 
    "unreachable": true
}

```
Same error comes while running a playbook too.

**Comment by hawksight** [comment]
So you've probably got an ssh issue., run the same ping command with `-vvvv` as the output suggests which might give you some hints as it should say where the connection fails.

Failing that, have you got the right ssh key to connect to the gcloud instances?
It really depends who / how they were built, but you might need to upload your ssh key to the instance. Or get the key that's already on the instance(s).
Try here for more info: https://cloud.google.com/compute/docs/instances/adding-removing-ssh-keys

If you have they key, perhaps add it with:
```
ssh-add <PATH_TO_KEY>
```

You can see current keys added with: `ssh-add -l`
